code/src/settings.c

```diff
@@ -56,6 +56,37 @@ s32 Settings_ApplyDamageMultiplier(GlobalContext* globalCtx, s32 changeHealth) {
 
     return modifiedChangeHealth;
 }
+//With the No Health Refill option on, full health refills from health upgrades and Bombchu Bowling are turned off, and fairies restore 3 hearts
+//Otherwise, they grant a full heal, and the default effect applies (full heal from bottle, 8 hearts on contact)
+u8 setFullHealthRestore(u8 setAmount) {
```

Comment:
we usually name functions in the patch folder like `[SystemName]_[FunctionName]` (this isnt 100% consistent in the code folder but its nice) so for example this function would be called `Settings_SetFullHealthRestore`, and similarly for the ones below this one